packages/react-link/src/components/Link/useLink.ts

```diff
@@ -1,27 +1,41 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { getNativeElementProps } from '@fluentui/react-utilities';
 import { useLinkState } from './useLinkState';
-import type { LinkProps, LinkState } from './Link.types';
+import type { LinkProps, LinkSlots, LinkState } from './Link.types';
 
 /**
  * Consts listing which props are shorthand props.
  */
-export const linkShorthandProps = [];
-
-const mergeProps = makeMergeProps<LinkState>({ deepMerge: linkShorthandProps });
+export const linkShorthandProps: Array<keyof LinkSlots> = ['root'];
 
 /**
  * Given user props, returns state and render function for a Link.
  */
-export const useLink = (props: LinkProps, ref: React.Ref<HTMLElement>, defaultProps?: LinkProps): LinkState => {
-  const state = mergeProps(
-    {
-      ref: useMergedRefs(ref, React.useRef<HTMLElement>(null)),
-      as: props.href ? 'a' : 'button',
+export const useLink = (props: LinkProps, ref: React.Ref<HTMLElement>): LinkState => {
+  const { as, disabled, disabledFocusable, href, inline, rel, target, secondary } = props;
+  const rootAs = as || (href ? 'a' : 'button');
+
+  const state: LinkState = {
+    // Props passed at the top-level
+    disabled,
+    disabledFocusable,
+    href,
+    inline,
+    rel,
+    secondary,
+    target,
```

Comment:
Some of these like `href`, `rel`, etc. are native props on the root slot. It could be confusing to also copy them to state?

Reply:
You are right, I have removed them.